app/assets/javascripts/school_administrator_dashboard/absences_dashboard/SchoolwideAbsences.js

```diff
@@ -7,9 +7,11 @@ import SchoolwideAttendance from './SchoolwideAttendance';
 class SchoolwideAbsences extends React.Component {
 
   render() {
+    const allAbsenceEvents = DashboardHelpers.absenceEvents(this.props.dashboardStudents);
     return (
       <SchoolwideAttendance
-        schoolAbsenceEvents = {DashboardHelpers.absenceEventsByDay(this.props.dashboardStudents)}
+        schoolAbsenceEvents = {allAbsenceEvents}
+        schoolAbsenceEventsByDay = {DashboardHelpers.eventsGroupedByDay(allAbsenceEvents)}
```

Comment:
I don't follow why this API is changing or why the same data in two different shapes are passed in, is this to try to re-use the grouping computation?

Reply:
@kevinrobinson it is, but frankly this may be more than is necessary. V1 of this feature was slow enough to render than everything got separated out wherever possible, but for readability's sake it might be better to bite the bullet and call this function a bunch of times in the next component. If you think that's better, I'll consolidate them.

Reply:
Got it.  Did you do this because of real performance issues you encountered making this change, or defensively to avoid those kinds of problems?  I think let's 🚢 this regardless, I'm  wondering on the current state.